geometric_kernels/spaces/product.py

```diff
@@ -312,6 +312,10 @@ def __init__(self, *spaces: DiscreteSpectrumSpace, num_eigen: int = 100):
     def dimension(self) -> int:
         return sum([space.dimension for space in self.sub_spaces])
 
+    def random(self, key, number):
+        # TBD
+        raise NotImplementedError
```

Comment:
This should just be a concatenation of random samples from each of the factors into a vector